pkg/internal/utils.go

```diff
@@ -225,64 +227,39 @@ func ExtractPodBindAnnotations(allocatedPod *core.Pod) map[string]string {
 	}
 }
 
-// PodSchedulingSpec comes from external, so need more Defaulting and Validation
-// when deserialization.
-func ExtractPodSchedulingSpec(pod *core.Pod) *si.PodSchedulingSpec {
+// ExtractPodSchedulingSpec extracts pod scheduling request from k8s pod request.
+// TODO: Need more defaulting and validation when deserialization.
+func ExtractPodSchedulingSpec(pod *core.Pod) *apiv2.PodSchedulingSpec {
 	// Consider all panics are BadRequestPanic.
 	defer AsBadRequestPanic()
 	errPfx := fmt.Sprintf("Pod annotation %v: ", si.AnnotationKeyPodSchedulingSpec)
 
-	podSchedulingSpec := si.PodSchedulingSpec{IgnoreK8sSuggestedNodes: true}
-
 	annotation := convertOldAnnotation(pod.Annotations[si.AnnotationKeyPodSchedulingSpec])
 	if annotation == "" {
 		panic(fmt.Errorf(errPfx + "Annotation does not exist or is empty"))
 	}
 
-	common.FromYaml(annotation, &podSchedulingSpec)
+	podSchedulingSpec := apiv2.PodSchedulingSpec{Version: "v2"}
+
+	generalSpec := apiv2.GeneralSpec{"version": "v1"}
+	common.FromYaml(annotation, &generalSpec)
+	switch generalSpec["version"] {
+	case "v1":
+		podSchedulingSpecV1 := si.PodSchedulingSpec{IgnoreK8sSuggestedNodes: true}
+		common.FromYaml(annotation, &podSchedulingSpecV1)
+		podSchedulingSpec.ConvertFromV1(&podSchedulingSpecV1)
```

Comment:
maybe we can keep v1 defaulting and validation here, instead of removing them

Reply:
updated